velox/exec/tests/utils/TpchBuilder.h

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/type/tests/FilterBuilder.h"
+#include "velox/type/tests/SubfieldFiltersBuilder.h"
+
+#if __has_include("filesystem")
+#include <filesystem>
+namespace fs = std::filesystem;
+#else
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+#endif
+
+namespace facebook::velox::exec::test {
+
+/// PlanContext contains the query plan and the sources.
+/// sources is a map where the key is the PlanNodeId and the value is a vector
+/// of file paths. All sources must be of the same file-format.
+struct PlanContext {
+  std::shared_ptr<core::PlanNode> plan_;
+  std::unordered_map<core::PlanNodeId, std::vector<std::string>> sources_;
+  dwio::common::FileFormat format_;
+};
+
+static int64_t toDate(std::string_view stringDate) {
+  Date date;
+  parseTo(stringDate, date);
+  return date.days();
+}
+
+/// This class builds TPC-H queries from a directory containing TPC-H data.
+/// Each table data must be placed in hive-style partitioning.
+/// Example: ls -R data/
+///  customer   lineitem
+///
+///  data/customer:
+///  customer1.parquet  customer2.parquet
+///
+///  data/lineitem:
+///  lineitem1.parquet  lineitem2.parquet  lineitem3.parquet
+
+/// The column names can vary. Additional columns may exist towards the end.
+/// The class uses standard names (example: l_returnflag) to build TPC-H plans.
+/// Since the column names in the file can vary, they are mapped to the standard
+/// names. Therefore, the order of the columns in the file is important and
+/// should be in the same order as in the TPC-H standard.
+class TpchBuilder {
+ public:
+  /// Read each data file and initialize row types and determine data paths for
+  /// each table.
+  /// @param dataPath path to the data files
+  /// @param format file format
+  void initialize(std::string dataPath, std::string format) {
```

Comment:
@majetideepak Would you move the implementation to a .cpp file for the sake of faster builds?

Reply:
I wanted to avoid an additional library(`velox-exec-test-util`) dependency by keeping it header only. I don't have a strong opinion here and can move it to a `.cpp` file if that is preferred.

Reply:
Let's move to .cpp